rllab/dynamics_randomization/mujoco_model_gen.py

```diff
@@ -0,0 +1,133 @@
+import atexit
+import queue
+from queue import Queue
+from threading import Event
+from threading import Thread
+
+from lxml import etree
+from mujoco_py import load_model_from_xml
+import numpy as np
+
+from rllab.dynamics_randomization import Distribution
+from rllab.dynamics_randomization import Method
+
+
+class MujocoModelGenerator:
+    """
+    A worker thread to produce to MuJoCo models with randomized dynamic
+    parameters, which are specified by the users of rllab with the class
+    Variations.
+    """
+
+    def __init__(self, file_path, variations):
+        """
+        Starts all the member fields of the class and the worker thread.
+        Parameters
+        ----------
+        file_path : string
+            The absolute path to the XML file that contains the MuJoCo
+            model.
+        variations: Variations
+            An list of Variation objects that indicate the dynamic parameters
+            to randomize in the XML file.
+        """
+        self._variations = variations
+        self._file_path = file_path
+        # Synchronized queue to store mujoco_models
+        self._models = Queue(maxsize=10)
+        # Worker Thread
+        self._worker_thread = Thread(
```

Comment:
were we able to show multithreading faster than single-threading on any system? if not, let's delete for now.

Reply:
No. After several tests, we found the multi-threading is slower than single-threading. Occasionally faster in milliseconds.